### PR TITLE
Add 4.1.0 release notes and documentation

### PR DESCRIPTION
**Pull request by brianhlin**
The environment documentation uses [tabbed code blocks](https://facelessuser.github.io/pymdown-extensions/extensions/superfences/#tabbed-fences) to show example configuration and resulting environment variables. 

For the APEL changes, we accidentally already applied those doc changes: https://github.com/htcondor/htcondor-ce/pull/250/files

### CODE REVIEW

**Comment by brianhlin** [comment]
We should also hold off on merging this until it's been released in the HTCondor repos

File: docs/batch-system-integration.md [hunk 1]
```diff
@@ -344,6 +344,77 @@ JOB_ROUTER_ENTRIES @=jre
...
+
+``` bash tab="Result"
+WN_SCRATCH_DIR=/nobackup/
+PILOT_COLLECTOR=collector.wisc.edu
+ACCOUNTING_GROUP=glow
+```
+
+!!!tip "Debugging job route environment expressions"
+    While constructing `set_default_pilot_job_env` expressions, try wrapping your expression in
+    [debug()](#debugging-routes) to help with any issues that may arise.
```
Comment (matyasselmeci) [inline docs/batch-system-integration.md]: Where does debug() output show up?

Comment (brianhlin) [reply docs/batch-system-integration.md]: In the JobRouterLog as mentioned in the linked section: https://htcondor-ce.readthedocs.io/en/latest/batch-system-integration/#debugging-routes

File: docs/releases.md [hunk 2]
```diff
@@ -51,14 +51,35 @@ HTCondor-CE 4 Version History
 This section contains release notes for each version of HTCondor-CE 4.
 Full HTCondor-CE version history can be found on [GitHub](https://github.com/htcondor/htcondor-ce/releases).
 
+### 4.1.0 ###
+
+[This release](https://github.com/htcondor/htcondor-ce/releases/tag/v4.1.0) includes the following new features:
+
+- **Add the ability to configure the environment of routed jobs:** Administrators may now add or override environment
```
Comment (timtheisen) [inline docs/releases.md]:
```suggestion
- **Added the ability to configure the environment of routed jobs:** Administrators may now add or override environment
```

**Pull request merged**